File: 2015/d03/main.py
text = input()


r, c = 0, 0


seen = set()
for d in text:
    if d == "^":
        r -= 1
    elif d == ">":
        c += 1
    elif d == "v":
        r += 1
    elif d == "<":
        c -= 1

    seen.add((r, c))


print(len(seen) + 1)

File: 2015/d03/main.rs
const INPUT: &str = include_str!("input");

fn main() {
    let (mut r, mut c) = (0, 0);
    let mut seen = std::collections::HashSet::new();

    for ch in INPUT.chars() {
        match ch {
            '^' => r -= 1,
            '>' => c += 1,
            'v' => r += 1,
            '<' => c -= 1,
            _ => {}
        };
        seen.insert((r, c));
    }

    println!("{}", seen.len() + 1);
}
